### PR TITLE
[WebIDL] add tests for is an array index abstract operation

### PR DESCRIPTION
**Pull request by tobie**
References https://github.com/heycam/webidl/pull/517.

### CODE REVIEW

**Comment by w3c-bots** [comment]
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/24084)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/339236240)

**Comment by TimothyGu** [comment]
I'd rather test this specifically for `document.all`, in which case a lot of these tests are already included:

https://github.com/w3c/web-platform-tests/blob/0c874a2e11885dc750902ae6c119f20affc6b49e/html/infrastructure/common-dom-interfaces/collections/htmlallcollection.html#L93-L100

**Comment by foolip** [comment]
Tests for document.all are in https://github.com/w3c/web-platform-tests/pull/9303

**Comment by foolip** [comment]
Don't review or merge it though, it's an automatic export!

**Comment by domenic** [comment]
@TimothyGu why is testing document.all specifically a good idea here? I'd rather test a less-crazy object in the WebIDL/ tests, leaving document.all tests to a dedicated file.

**Comment by TimothyGu** [comment]
@Domenic It’s not, I agree with you. But this PR currently tests `document.all`.

**Comment by domenic** [comment]
Hmm, I guess I misunderstood your comment then, OK.

**Comment by tobie** [comment]
I’m fine testing using another object. Which one would you suggest that allows testing the same things?

**Comment by domenic** [comment]
Anything with an array index, right? I guess you'd ideally test all the call sites: get/set/define/delete. Maybe one object with an indexed getter only, and one object with an indexed setter? So like, NodeList and Storage?

File: WebIDL/ecmascript-binding/legacy-platform-object.html [hunk 1]
```diff
@@ -188,4 +192,20 @@
                           {value: option0, writable: true, enumerable: true,
                            configurable: true});
 }, "Test [[DefineOwnProperty]] with indexed property setter support.");
+
+test(function() {
+    // https://heycam.github.io/webidl/#is-an-array-index
+    assert_equals(document.all["4892.123"].id, "4892.123",
+        "Numbers with decimal points are not treated as array indices");
+    assert_equals(document.all["-0"], undefined, "'-0' is special cased");
```
Comment (foolip) [inline WebIDL/ecmascript-binding/legacy-platform-object.html]: This is tested by https://github.com/w3c/web-platform-tests/pull/9303. Testing some other object that `document.all` would be nice.

**Comment by foolip** [comment]
@tobie is this something you'd like to finish if I do review, or should we close it?

**Pull request closed without merging**